File: src/main/java/co/gatedaccess/web/config/AuditorAwareImpl.java
package co.gatedaccess.web.config;

import org.springframework.data.domain.AuditorAware;

import java.util.Date;
import java.util.Optional;

public class AuditorAwareImpl implements AuditorAware<Date> {
    @Override
    public Optional<Date> getCurrentAuditor() {
        return Optional.of(new Date());
    }
}

File: src/main/kotlin/co/gatedaccess/web/config/AuditorAwareImpl.kt
package co.gatedaccess.web.config

import org.springframework.data.domain.AuditorAware
import java.util.*

class AuditorAwareImpl : AuditorAware<Date> {
    override fun getCurrentAuditor(): Optional<Date> {
        return Optional.of(Date())
    }
}
